var/spack/repos/builtin/packages/libunwind/package.py

```diff
@@ -32,6 +32,8 @@ class Libunwind(AutotoolsPackage):
             description='Support zlib compressed symbol tables '
             '(1.5 and later).')
 
+    patch('gcc.patch', when=':@1.6 %gcc@10.0:')
```

Comment:
```suggestion
    patch('gcc.patch', when='@:1.6 %gcc@10.0:')
```
This fixes the failing unit tests.

Reply:
Apart from this, the patch only applies cleanly to 1.4 (the older versions fail and 1.5 already has it applied). I would suggest to simply do something like this (so we do not have to carry the large patch):
```python
def setup_build_environment(self, env):
    # https://github.com/libunwind/libunwind/pull/166
    if self.spec.satisfies('%gcc@10:'):
        env.set('CFLAGS', '-fcommon')
```